opal/corelib/kernel.rb

```diff
@@ -205,8 +205,13 @@ def extend(*mods)
       for (var i = mods.length - 1; i >= 0; i--) {
         var mod = mods[i];
 
-        #{`mod`.append_features `singleton`};
-        #{`mod`.extended self};
+        if (mod.$$is_module) {
+          #{`mod`.append_features `singleton`};
+          #{`mod`.extended self};
+        }
+        else {
+          throw Opal.TypeError.$new("wrong argument type Class (expected Module)");
```

Comment:
1. I think you can raise a Ruby-level exception here, this code depends on internal implementation of Opal
2. If an object is not module, it doesn't mean that it's a class.
``` ruby
raise TypeError, "wrong argument type #{`mod`.class}"
```